app-testing/tests/protocol_upload_test.py

```
@@ -0,0 +1,127 @@
+"""Test the initial state the application with various setups."""
+import logging
+import os
+from pathlib import Path
+import platform
+import time
+import pytest
+from typing import Dict
+from selenium import webdriver
+from pytest import FixtureRequest
+from selenium.webdriver.chrome.options import Options
+
+from src.menus.left_menu import LeftMenu
+from src.pages.protocol_upload import ProtocolUpload
+from src.pages.deck_calibrate import DeckCalibration
+from src.resources.ot_application import OtApplication
+from src.pages.overview import Overview
+from src.pages.robot_page import RobotPage
+from src.pages.robot_calibration import RobotCalibration
+from src.pages.module_setup import ModuleSetup
+from src.pages.labware_setup import LabwareSetup
+from src.menus.robots_list import RobotsList
+from src.menus.protocol_file import ProtocolFile
+from src.driver.drag_drop import drag_and_drop_file
+
+logger = logging.getLogger(__name__)
+
+
+def test_protocol_upload(
+    chrome_options: Options,
+    test_json_protocols: Dict[str, Path],
+    request: FixtureRequest,
+) -> None:
```

This test requires a robot running or the dev emulator correct?   Including the below validates it is running so the test will fail fast if it is not.
````python
    robot = OtRobot()
    # expecting docker emulated robot
    assert robot.is_alive(), "is a robot available?"
````